docs/30-data-valuation.rst

```diff
@@ -23,118 +23,380 @@ interest, but a function of three factors:
+it returns as exact a value as the utility function allows (see what this means
+in :ref:`problems of data values`).
+
 The value $v$ of the $i$-th sample in dataset $D$ wrt. utility $u$ is computed
 as a weighted sum of its marginal utility wrt. every possible coalition of
 training samples within the training set:
 
 $$v_u(x_i) = \frac{1}{n} \sum_{S \subseteq D \setminus \{x_i\}} \binom{n-1}{ | S | }^{-1} [u(S \cup \{x_i\}) − u(S)] ,$$
 
+.. code-block:: python
+
+   from pydvl.value import compute_shapley_value
+   utility = Utility(...)
+   df = compute_shapley_values(utility, mode="combinatorial_exact")
+
+The return value `df` is a
+`pandas DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_
+with the values. Please refer to the documentation in :mod:`pydvl.value.shapley`
+for more information.
+
+Monte Carlo Combinatorial Shapley
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 Because the number of subsets $S \subseteq D \setminus \{x_i\}$ is
 $2^{ | D | - 1 }$, one typically must resort to approximations. The simplest
-one is done via Monte Carlo sampling of the powerset $\mathcal{P}(D)$.
+one is done via Monte Carlo sampling of the powerset $\mathcal{P}(D)$. In pyDVL
+this simple technique is called "Monte Carlo Combinatorial". The method has very
+poor converge rate and others are preferred, but if desired, usage follows the
+same pattern:
+
+.. code-block:: python
+
+   from pydvl.utils import Dataset, Utility
+   from pydvl.value.shapley import compute_shapley_values
+   model = ...
+   data = Dataset(...)
+   utility = Utility(model, data)
+   df = compute_shapley_values(utility, mode="combinatorial_montecarlo")
 
-However, an equivalent formulation of the expression above is typically used
-which uses permutations over indices instead of subsets:
+The DataFrame returned by Monte Carlo methods will usually contain approximate
+standard errors as an additional column.
+
+
+Owen sampling
+^^^^^^^^^^^^^
+
+**Owen Sampling** (:footcite:t:`okhrati_multilinear_2021`) is a practical
+algorithm based on the combinatorial definition. It uses a continuous extension
+of the utility from $\{0,1\}^n$, where a 1 in position $i$ means that sample
+$x_i$ is used to train the model, to $[0,1]^n$. The ensuing expression for
+Shapley value uses integration instead of discrete weights:
+
+$$v_u(i) = \int_0^1 \mathbb{E}_{S \sim P_q(D_{\backslash \{ i \}})} [u(S \cup {i}) - u(S)].$$
+
+Using Owen sampling follows the same pattern as every other method for Shapley
+values in pyDVL. First construct the dataset and utility, then call
+:func:`~pydvl.value.shapley.compute_shapley_values`:
+
+.. code-block:: python
+
+   from pydvl.utils import Dataset, Utility
+   from pydvl.value.shapley import compute_shapley_values
+   model = ...
+   dataset = Dataset(...)
+   utility = Utility(data, model)
+   df = compute_shapley_values(
+           u=utility, mode="owen", max_iterations=4, max_q=200
+       )
```

Comment:
Minor nit: The indentation is off

Reply:
black seems to think it's ok. Otherwise the tests wouldn't pass, right?

Reply:
I don't think black checks rst files. We could consider using [blacken-docs](https://pypi.org/project/blacken-docs/) to do that.

Reply:
I really need to learn to pause before writing the first that comes to my mind. Will fix